x-pack/solutions/security/plugins/elastic_assistant/server/lib/attack_discovery/schedules/types.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DefaultAlert } from '@kbn/alerts-as-data-utils';
+import { RuleExecutorOptions, RuleType, RuleTypeState } from '@kbn/alerting-plugin/server';
+import {
+  ALERT_ATTACK_DISCOVERY_ALERTS_CONTEXT_COUNT,
+  ALERT_ATTACK_DISCOVERY_ALERT_IDS,
+  ALERT_ATTACK_DISCOVERY_API_CONFIG,
+  ALERT_ATTACK_DISCOVERY_DETAILS_MARKDOWN,
+  ALERT_ATTACK_DISCOVERY_ENTITY_SUMMARY_MARKDOWN,
+  ALERT_ATTACK_DISCOVERY_MITRE_ATTACK_TACTICS,
+  ALERT_ATTACK_DISCOVERY_REPLACEMENTS,
+  ALERT_ATTACK_DISCOVERY_SUMMARY_MARKDOWN,
+  ALERT_ATTACK_DISCOVERY_TITLE,
+  ALERT_ATTACK_DISCOVERY_USERS,
+} from './fields';
+
+export type AttackDiscoveryAlert = DefaultAlert & {
```

Comment:
These are the basic types that we gonna use for executor and rule registration. Executor definition will be updated once we define attack discovery schedule parameters.